src/main/java/com/minecolonies/client/gui/WindowBuildTool.java

```diff
@@ -420,7 +429,8 @@ private void styleClicked(@NotNull Button button)
      */
     private void levelClicked(Button button)
     {
-        int maxLevel = Schematics.getMaxLevelForHut(hutDec.get(hutDecIndex));
+        List<String> styles = getStyles();
+        int maxLevel = Schematics.getMaxLevelForHut(hutDec.get(hutDecIndex),styles.get(styleIndex));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'maxLevel' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)